# Lab 2 Rohit Jayakumar Nair

Q1. Download the dataset at https://raw.githubusercontent.com/uc-python/intro-pythondatasci/master/data/planes.csv and load it into Python as a DataFrame. Store it in a variable called df.


In [57]:
import pandas as pd
df = pd.read_csv('C:/Users/rohit/Downloads/intro-python-datasci-master/intro-python-datasci-master/data/planes.csv')

Q2. Print the first few rows of the DataFrame using the .head() method of df.

In [4]:
df.head()

tailnum    year                     type      manufacturer      model  \
0  N10156  2004.0  Fixed wing multi engine           EMBRAER  EMB-145XR   
1  N102UW  1998.0  Fixed wing multi engine  AIRBUS INDUSTRIE   A320-214   
2  N103US  1999.0  Fixed wing multi engine  AIRBUS INDUSTRIE   A320-214   
3  N104UW  1999.0  Fixed wing multi engine  AIRBUS INDUSTRIE   A320-214   
4  N10575  2002.0  Fixed wing multi engine           EMBRAER  EMB-145LR   

   engines  seats  speed     engine  
0        2     55    NaN  Turbo-fan  
1        2    182    NaN  Turbo-fan  
2        2    182    NaN  Turbo-fan  
3        2    182    NaN  Turbo-fan  
4        2     55    NaN  Turbo-fan

Q3. Take a look at a few attributes of df: .shape, .index and .columns. Do the contents of these attributes give you any information about the DataFrame?

In [5]:
df.shape

(3322, 9)

In [6]:
df.index

RangeIndex(start=0, stop=3322, step=1)

In [7]:
df.columns

Index(['tailnum', 'year', 'type', 'manufacturer', 'model', 'engines', 'seats',
       'speed', 'engine'],
      dtype='object')

The df.shape gives returns a tuple representing the dimensionality of the DataFrame.

The df.index gives us row labels(index) of the DataFrame.

The df.columns returns the column labels of the DataFrame.

Q4. Check the data types of the columns of df. Hint: We did this during lecture in one of the
early slides.

In [11]:
df.dtypes

tailnum          object
year            float64
type             object
manufacturer     object
model            object
engines           int64
seats             int64
speed           float64
engine           object
dtype: object

Q5. Select just the year column using df['year']. Store the result in a variable called
year. Print it out

In [14]:
year = df['year']

In [15]:
year.head()

0    2004.0
1    1998.0
2    1999.0
3    1999.0
4    2002.0
Name: year, dtype: float64

Q6. What is the type of year?


In [16]:
type(year)

pandas.core.series.Series

Q7. Select both the manufacturer and model columns and store the result in a variable called plane_types. Print it out.


In [18]:
plane_types = df[['manufacturer','model']]

In [19]:
plane_types.head()

manufacturer      model
0           EMBRAER  EMB-145XR
1  AIRBUS INDUSTRIE   A320-214
2  AIRBUS INDUSTRIE   A320-214
3  AIRBUS INDUSTRIE   A320-214
4           EMBRAER  EMB-145LR

Q8. What is the type of plane_types? Why is this different from #6?


In [20]:
type(plane_types)

pandas.core.frame.DataFrame

Year is a series, which is a type of list in pandas which can take integer values, string values, double values etc.
Series can only contain single list with index


Whereas plane_types which is a dataframe can be made of more than one series or we can say that a dataframe is a collection of series that can be used to analyse the data.

Q9. Slice the row at index 105. What year was this plane made?

In [22]:
year[105]

2004.0

Q10. Filter down the dataset to only planes made by Boeing. How many rows are left? Hint:
Use .shape to get a row count.

In [65]:
df3= df[df['manufacturer'] == 'BOEING']

In [66]:
df3.shape

(1630, 9)

Q11. In the filtered data from #10, what do you notice about the indexes? Call the .reset_index() method on this data and see what changes.


In [37]:
df.index

Int64Index([  34,   49,   50,   51,   52,   53,   55,   69,   72,   73,
            ...
            3301, 3303, 3305, 3307, 3309, 3311, 3313, 3315, 3317, 3319],
           dtype='int64', length=1630)

In [38]:
df.reset_index()

index tailnum    year                     type manufacturer    model  \
0        34  N11206  2000.0  Fixed wing multi engine       BOEING  737-824   
1        49  N1200K  1998.0  Fixed wing multi engine       BOEING  767-332   
2        50  N1201P  1998.0  Fixed wing multi engine       BOEING  767-332   
3        51  N12109  1994.0  Fixed wing multi engine       BOEING  757-224   
4        52  N12114  1995.0  Fixed wing multi engine       BOEING  757-224   
...     ...     ...     ...                      ...          ...      ...   
1625   3311  N994AT  2002.0  Fixed wing multi engine       BOEING  717-200   
1626   3313  N995AT  2002.0  Fixed wing multi engine       BOEING  717-200   
1627   3315  N996AT  2002.0  Fixed wing multi engine       BOEING  717-200   
1628   3317  N997AT  2002.0  Fixed wing multi engine       BOEING  717-200   
1629   3319  N998AT  2002.0  Fixed wing multi engine       BOEING  717-200   

      engines  seats  speed     engine  
0           2    149    NaN  Turbo-fan  
1           2    330    NaN  Turbo-fan  
2           2    330    NaN  Turbo-fan  
3           2    178    NaN  Turbo-jet  
4           2    178    NaN  Turbo-jet  
...       ...    ...    ...        ...  
1625        2    100    NaN  Turbo-fan  
1626        2    100    NaN  Turbo-fan  
1627        2    100    NaN  Turbo-fan  
1628        2    100    NaN  Turbo-fan  
1629        2    100    NaN  Turbo-fan  

[1630 rows x 10 columns]

Q12. How many planes in the data have fewer than 40 seats?

In [49]:
df2 = df[df['seats'] < 40]

In [51]:
df2.shape

(122, 9)

Q13. What is the minimum number of seats of any plane in the data? Hint: This is a good time
to use summary functions.


In [63]:
df.describe()

year      engines        seats       speed
count  3252.000000  3322.000000  3322.000000   23.000000
mean   2000.484010     1.995184   154.316376  236.782609
std       7.193425     0.117593    73.654974  149.759794
min    1956.000000     1.000000     2.000000   90.000000
25%    1997.000000     2.000000   140.000000  107.500000
50%    2001.000000     2.000000   149.000000  162.000000
75%    2005.000000     2.000000   182.000000  432.000000
max    2013.000000     4.000000   450.000000  432.000000

Q14. What is the minimum number of seats in any plane made by Boeing?


In [67]:
df3.describe()

year      engines       seats  speed
count  1603.000000  1630.000000  1630.00000    0.0
mean   2000.144105     2.001227   175.18773    NaN
std       7.008365     0.049538    59.46881    NaN
min    1965.000000     2.000000   100.00000    NaN
25%    1996.000000     2.000000   140.00000    NaN
50%    2000.000000     2.000000   149.00000    NaN
75%    2005.000000     2.000000   178.00000    NaN
max    2013.000000     4.000000   450.00000    NaN

The min number of seats in any plane made by Boeing is 100.

Q15. Create a new column in df called seats_per_engine. It is defined as number of seats
divided by number of engines.


In [73]:
df['seats_per_engine'] = df['seats']/df['engines']

In [74]:
df.head()

tailnum    year                     type      manufacturer      model  \
0  N10156  2004.0  Fixed wing multi engine           EMBRAER  EMB-145XR   
1  N102UW  1998.0  Fixed wing multi engine  AIRBUS INDUSTRIE   A320-214   
2  N103US  1999.0  Fixed wing multi engine  AIRBUS INDUSTRIE   A320-214   
3  N104UW  1999.0  Fixed wing multi engine  AIRBUS INDUSTRIE   A320-214   
4  N10575  2002.0  Fixed wing multi engine           EMBRAER  EMB-145LR   

   engines  seats  speed     engine  seats_per_engine  
0        2     55    NaN  Turbo-fan              27.5  
1        2    182    NaN  Turbo-fan              91.0  
2        2    182    NaN  Turbo-fan              91.0  
3        2    182    NaN  Turbo-fan              91.0  
4        2     55    NaN  Turbo-fan              27.5

Q16. What is the average (mean) number of seats per engine in the data?


In [75]:
df.describe()

year      engines        seats       speed  seats_per_engine
count  3252.000000  3322.000000  3322.000000   23.000000       3322.000000
mean   2000.484010     1.995184   154.316376  236.782609         77.064996
std       7.193425     0.117593    73.654974  149.759794         36.589602
min    1956.000000     1.000000     2.000000   90.000000          0.500000
25%    1997.000000     2.000000   140.000000  107.500000         70.000000
50%    2001.000000     2.000000   149.000000  162.000000         74.500000
75%    2005.000000     2.000000   182.000000  432.000000         91.000000
max    2013.000000     4.000000   450.000000  432.000000        200.000000

The mean seats per engine is 77.06

Q17. Create a subset of the data, containing only planes manufactured by Cessna. Select all
the columns except manufacturer (since we know they're all Cessna). Export the data as
a CSV called "cessna_planes.csv".

In [76]:
df_cessna = df[df['manufacturer'] == 'CESSNA']

In [78]:
df_cessna.shape

(9, 10)

In [80]:
df_cessna.drop(columns=['manufacturer'])

tailnum    year                      type       model  engines  seats  \
424   N201AA  1959.0  Fixed wing single engine         150        1      2   
427   N202AA  1980.0   Fixed wing multi engine        421C        2      8   
893   N364AA  1973.0   Fixed wing multi engine        310Q        2      6   
1027  N378AA  1963.0  Fixed wing single engine        172E        1      4   
1476  N519AA  1979.0   Fixed wing multi engine         550        2      8   
1480  N519MQ  1983.0  Fixed wing single engine       A185F        1      6   
1725  N575AA  1963.0  Fixed wing single engine  210-5(205)        1      6   
1883  N621AA  1975.0  Fixed wing single engine        172M        1      4   
2309  N737MQ  1977.0  Fixed wing single engine        172N        1      4   

      speed         engine  seats_per_engine  
424    90.0  Reciprocating               2.0  
427    90.0  Reciprocating               4.0  
893   167.0  Reciprocating               3.0  
1027  105.0  Reciprocating               4.0  
1476    NaN      Turbo-fan               4.0  
1480  127.0  Reciprocating               6.0  
1725    NaN  Reciprocating               6.0  
1883  108.0        4 Cycle               4.0  
2309  105.0  Reciprocating               4.0

In [81]:
df_cessna.to_csv(r'C:/Users/rohit/Downloads/intro-python-datasci-master/intro-python-datasci-master/data/cessna_planes.csv')